aform/src/components/form/ADropdown.vue

```diff
@@ -29,86 +28,89 @@
 </template>
 
 <script setup lang="ts">
-import { onMounted, onUnmounted, ref /* useTemplateRef */ } from 'vue'
-
-const { label, items, isAsync } = defineProps<{
+import { ref, useTemplateRef } from 'vue'
+import { vOnClickOutside } from '@vueuse/components'
+const { label, items, isAsync, filterFunction } = defineProps<{
 	label: string
 	items?: string[]
 	isAsync?: boolean
+	filterFunction?: (search: string) => Promise<string[]>
```

Comment:
A couple of things:
- Should the `filterFunction` be allowed to be sync as well? The filter function you've added in the test is really just a sync function, so maybe we support both?
- Should the items also be passed to the `filterFunction`? So that the user has context on the current list of items.

Reply:
> Regarding passing items to the function, I don’t see a clear use case where it’s necessary. Do you think in some case?

Hmm, that's true too. I suppose if the context on the items is available in the component itself, then we don't need to worry about passing in the items.